docs/user-guide/timeouts/timeouts-main.md

```diff
@@ -0,0 +1,331 @@
+that Certora Prover is solving (being undecidable in general).
+
+#### Running with Yices
+
+A different choice of solver sometimes helps. The *Yices* SMT solver ([home
+page](https://yices.csl.sri.com/)) is not used by default since it is not
+compatible with our default hashing scheme. 
+
+The following setting sets a hashing scheme that is compatible with Yices. Since
+Yices is in the default portfolio, it is included automatically then.
+
+```
+--prover_args "-smt_hashingScheme plainInjectivity"
+```
+
+Optionally, we can further prioritize the usage of Yices by decreasing the size
+of the solver portfolio. With the `-solvers` option set as follows, Certora
+Prover will run only CVC5 and Yices. Furthermore, we can make Certora Prover use
+the ordering given in the `-solvers` option for prioritizing solvers using the
+`-smt_overrideSolvers` option.
+
+```
+--prover_args "-solvers [yices, cvc5] -smt_overrideSolvers true"
+```
+
+
+(modular_verification)=
+## Modular verification
+
+Especially for large code bases, but also for instance when there are parts with
+particularly complex behavior, it helps to modularize the verification process.
+In the following we elaborate on modularization techniques that can help
+preventing timeouts.
+
+
+### "Sanity" Rules
+
+For isolating the timeout reason, it can be useful to verify the code with
+respect to a trivial specification. This, to some extent, rules out the
+specification as the source of complexity.
+
+Sanity rules are such trivial specifications. For documentation on them, see
+{ref}`sanity <built-in-sanity>` and {ref}`deep sanity <built-in-deep-sanity>`. 
+
+
+(library_timeouts)=
+### Library-based systems
+
+Some of the systems we have are based on multiple library contracts which
+implement the business logic. They also forward storage updates to a single
+external contract holding the storage.
+
+In these systems, it’s sensible to split the verification so as each library is
+operated on an individual basis.
+
+If you encounter timeouts when trying to verify the main entry point contract to
+the system, check the impact of the libraries on the verification by summarizing
+all external library (delegate) calls as `NONDET`, using the option
+`summarizeExtLibraryCallsAsNonDetPreLinking` as follows:
+
+```
+certoraRun ... --prover_args '-summarizeExtLibraryCallsAsNonDetPreLinking true'
+```
+
+```{note}
+This option is only applied for _external_ library calls, or `delegatecall`s.
+Internal calls are automatically inlined by the Solidity compiler and are 
+subject to summarizations specified in the spec file's `methods` block.
+```
+
+Alternatively, if you wish to apply a "catch-all" summary for all the methods of
+a specific library, you can write in the methods block of the spec:
+
+```
+methods {
+    function MyBigLibrary._ external => NONDET;
+    function MyBigLibrary._ internal => NONDET;
+}
+```
+
+The above snippet has the effect of summarizing as `NONDET` all external calls
```

Comment:
Do we have a dedicated section on summarisation in our doc? If yes, perhaps link it?

Reply:
good point, added a sentence with a link